modules/fedimint-ecash-migration-server/src/lib.rs

```diff
@@ -227,20 +435,321 @@ impl ServerModule for EcashMigration {
+                    module.get_transfer_id(&mut context.dbtx().into_nc(), request)
+                        .await
+                        .ok_or_else(|| ApiError::not_found("Transfer ID not found for out point".to_owned()))
+                }
+            },
+            // Idempotent to allow retries for fault tolerance
+            api_endpoint! {
+                UPLOAD_KEY_SET_ENDPOINT,
+                ApiVersion::new(0, 0),
+                async |module: &EcashMigration, context, request: UploadKeySetRequest| -> () {
+                    check_auth(context, request.transfer_id, &request).await?;
+                    module.upload_key_set(&mut context.dbtx().into_nc(), request.transfer_id, request.tier_keys)
+                        .await
+                }
+            },
+            api_endpoint! {
+                UPLOAD_SPEND_BOOK_BATCH_ENDPOINT,
+                ApiVersion::new(0, 0),
+                async |module: &EcashMigration, context, request: UploadSpendBookBatchRequest| -> UploadSpendBookBatchResponse {
+                    check_auth(context, request.transfer_id, &request).await?;
+                    module.upload_spend_book_batch(&mut context.dbtx().into_nc(), request.transfer_id, request.entries)
+                        .await
+                }
+            },
+            // Used to resume aborted uploads
+            api_endpoint! {
+                GET_UPLOADED_SPEND_BOOK_ENTRIES_ENDPOINT,
+                ApiVersion::new(0, 0),
+                async |module: &EcashMigration, context, request: TransferId| -> u64 {
+                    module.get_uploaded_spend_book_entries(&mut context.dbtx().into_nc(), request)
+                        .await
+                }
+            },
+            api_endpoint! {
+                CHECK_SPEND_BOOK_HASH_ENDPOINT,
+                ApiVersion::new(0, 0),
+                async |module: &EcashMigration, context, request: TransferId| -> SpendBookHash {
+                    // TODO: check auth
+                    module.check_spend_book_hash(&mut context.dbtx().into_nc(), request)
+                        .await
+                }
+            },
+            api_endpoint! {
+                GET_TRANSFER_STATUS_ENDPOINT,
+                ApiVersion::new(0, 0),
+                async |module: &EcashMigration, context, request: GetTransferStatusRequest| -> GetTransferStatusResponse {
+                    module.get_transfer_status(&mut context.dbtx().into_nc(), request.transfer_id)
+                        .await
+                }
+            },
+        ]
     }
 }
 
 impl EcashMigration {
-    /// Create new module instance
-    pub fn new(cfg: EcashMigrationConfig) -> EcashMigration {
-        EcashMigration { cfg }
+    /// Returns the transfer ID of the transfer created by the output
+    async fn get_transfer_id(
+        &self,
+        dbtx: &mut DatabaseTransaction<'_>,
+        out_point: OutPoint,
```

Comment:
I still don't know what is this `OutPoint` and I don't like it anyway. Couldn't we key requests like this on spend_book_hash or something? That should be unique already.